docs/world api.md

```diff
@@ -530,74 +439,104 @@ class MyGameWorld(World):
 * `generate_early(self)`
-  called per player before any items or locations are created. You can set properties on your world here. Already has
-  access to player options and RNG. This is the earliest step where the world should start setting up for the current
-  multiworld as any steps before this, the multiworld itself is still getting set up
+  called per player before any items or locations are created. You can set properties on your
+  world here. Already has access to player options and RNG. This is the earliest step where the world should start
+  setting up for the current multiworld, as any steps before this the multiworld itself is still getting set up.
 * `create_regions(self)`
-  called to place player's regions and their locations into the MultiWorld's regions list. If it's
-  hard to separate, this can be done during `generate_early` or `create_items` as well.
+  called to place player's regions and their locations into the MultiWorld's regions list.
+  If it's hard to separate, this can be done during `generate_early` or `create_items` as well.
 * `create_items(self)`
-  called to place player's items into the MultiWorld's itempool. After this step all regions and items have to be in
-  the MultiWorld's regions and itempool, and these lists should not be modified afterwards.
+  called to place player's items into the MultiWorld's itempool. After this step all regions
+  and items have to be in the MultiWorld's regions and itempool, and these lists should not be modified afterward.
 * `set_rules(self)`
-  called to set access and item rules on locations and entrances. 
-  Locations have to be defined before this, or rule application can miss them.
+  called to set access and item rules on locations and entrances.
 * `generate_basic(self)`
-  called after the previous steps. Some placement and player specific
-  randomizations can be done here.
+  called after the previous steps. Player specific randomization that does not affect logic can be done here.
 * `pre_fill(self)`, `fill_hook(self)` and `post_fill(self)` are called to modify item placement
-  before, during and after the regular fill process, before `generate_output`.
-  If items need to be placed during pre_fill, these items can be determined
-  and created using `get_prefill_items`
-* `generate_output(self, output_directory: str)` that creates the output
-  files if there is output to be generated. When this is
-  called, `self.multiworld.get_locations(self.player)` has all locations for the player, with
-  attribute `item` pointing to the item.
-  `location.item.player` can be used to see if it's a local item.
+  before, during and after the regular fill process; all finishing before `generate_output`. Any items that need to be
+  placed during `pre_fill` should not exist in the itempool, and if any items that need to be filled this way, but need
+  to be in state while you fill other items, they can be returned from `get_prefill_items`.
```

Comment:
Could also change "return" to a different verb like "read" or "acquired", but I think "returned" makes sense
```suggestion
  to be in state while you fill other items, they can be returned with `get_prefill_items`.
```

Reply:
not exactly. `get_prefill_items` returns a `List[Item]` and it gets called specifically by `get_all_state`, not the world itself. there probably could be better wording but the suggestion here is slightly incorrect.